src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -352,7 +352,7 @@ public async Task AddRewindDataAsync(IList<UnspentOutputs> unspentOutputs, Chain
                     {
                         // We'll need to restore the original outputs.
                         UnspentOutputs clone = unspent.Clone();
-                        clone.Outputs = original.OriginalOutputs;
+                        clone.Outputs = original.UnspentOutputs.Outputs;
```

Comment:
Should this not be using toarray in order to have a clone?

Reply:
I debugged the issue, and @fenix2222 sent the code fix over so I could test locally.  Happy to change, but would like confirmation from @fenix2222 first before changing.